packages/server/lib/factory/IMAInternalFactory.js

```diff
@@ -0,0 +1,258 @@
+  appFactory,
+  languageLoader,
+}) {
+  const GLOBAL = {
+    APP_MAIN: 'appMain',
+    DUMMY_APP: 'dummyApp',
+  };
+
+  function _createDummyApp({ environment }) {
+    // TODO IMA@18 doc dummy APP
+    // BETTER 404 detection
+    const event = createEvent('createDummyApp', {
+      context: {},
+      environment,
+      res: {
+        app: {},
+        headersSent: false,
+        locals: {
+          language: 'en',
+        },
+        append() {},
+        attachment() {},
+        clearCookie() {},
+        download() {},
+        end() {},
+        format() {},
+        json() {},
+        jsonp() {},
+        links() {},
+        location() {},
+        redirect() {},
+        render() {},
+        send() {},
+        sendFile() {},
+        sendStatus() {},
+        set() {},
+        setHeader() {},
+        type() {},
+        get() {},
+        getHeader() {},
+      },
+      req: {
+        app: {},
+        baseUrl: '',
+        body: {},
+        cookies: {},
+        fresh: true,
+        hostname: '',
+        headers: {},
+        ip: '127.0.0.1',
+        ips: [],
+        method: 'GET',
+        originalUrl: '/',
+        params: {},
+        path: '',
+        protocol: 'https',
+        query: {},
+        route: {},
+        secure: true,
+        signedCookies: {},
+        stale: false,
+        subdomains: [],
+        xhr: true,
+        accepts() {},
+        acceptsCharsets() {},
+        acceptsEncodings() {},
+        acceptsLanguages() {},
+        get() {},
+        is() {},
+        range() {},
+      },
+    });
+
+    const bootConfig = createBootConfig(event);
+    const appMain = serverGlobal.get(GLOBAL.APP_MAIN);
+    event.context.app = appMain.ima.createImaApp();
+
+    emitter.emit(Event.CreateImaApp, event);
+
+    Object.assign(
+      bootConfig,
+      appMain.getInitialAppConfigFunctions(),
+      appMain.ima.getInitialPluginConfig(),
+      appMain.ima.getInitialImaConfigFunctions()
+    );
+    event.context.app.bootstrap.run(bootConfig);
+
+    return event.context.app;
+  }
+
+  function _getRouteInfo({ req, res }) {
+    let routeInfo = null;
+
+    if (!serverGlobal.has(GLOBAL.DUMMY_APP)) {
+      return routeInfo;
+    }
+
+    const dummyApp = serverGlobal.get(GLOBAL.DUMMY_APP);
+    const {
+      protocol,
+      host,
+      path: urlPath,
+      root,
+      languagePartPath,
+    } = res.locals;
+
+    dummyApp.oc.get('$Request').init(req);
+    dummyApp.oc.get('$Response').init(res);
+    dummyApp.oc.get('$Router').init({
+      $Protocol: protocol,
+      $Host: host,
+      $Path: urlPath,
+      $Root: root,
+      $LanguagePartPath: languagePartPath,
+    });
+
+    try {
+      routeInfo = dummyApp.oc.get('$Router').getCurrentRouteInfo();
+    } catch (e) {
+      console.warn('Failed to retrieve current route info', {
+        error: e,
+      });
+    }
+
+    return routeInfo;
+  }
+
+  function _addImaToResponse({ req, res }) {
+    let routeName = 'other';
```

Comment:
I know it was already there... but what's the point of `'other'` ?

Reply:
only for keeping routeName as string..